core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -722,6 +722,9 @@ class SparkContext(config: SparkConf) extends Logging {
     }
     appStatusSource.foreach(_env.metricsSystem.registerSource(_))
     _plugins.foreach(_.registerMetrics(applicationId))
+
+    new CallerContext("DRIVER", config.get(APP_CALLER_CONTEXT),
+      Option(applicationId), applicationAttemptId).setCurrentContext()
```

Comment:
`applicationId` won't be `null` here

```suggestion
      Some(applicationId), applicationAttemptId).setCurrentContext()
```

Reply:
Done